### PR TITLE
[SYCLomatic] Update 'dpcpp' command with 'icpx -fsycl' because 'dpcpp' command will be deprecated in 2023.0

### PR DESCRIPTION
**Pull request by tomflinda**
Signed-off-by: chenwei.sun <chenwei.sun@intel.com>

### CODE REVIEW

**Review by ziranzha — APPROVED** [review]
LGTM

File: clang/lib/DPCT/GenMakefile.cpp [hunk 1]
```diff
@@ -410,7 +410,7 @@ genMakefile(clang::tooling::RefactoringTool &Tool, StringRef OutRoot,
                                   "_FLAG_" + std::to_string(Idx);
         OS << buildString("$(", ObjStrName, "):$(", SrcStrName, ")\n");
 
-        // Only apply dpcpp compiler to the files which are originally built
+        // Only apply 'icpx' compiler to the files which are originally built
```
Comment (zhimingwang36) [inline clang/lib/DPCT/GenMakefile.cpp]: Line 413,  the  meaning of the status is changed when using icpx.

File: clang/lib/DPCT/GenMakefile.cpp [hunk 2]
```diff
@@ -96,8 +96,8 @@ static void getCompileInfo(
           // Set the target name
           TargetName = Obj;
           IsTargetName = false;
-          Tool = "dpcpp -o"; // Record the tool that generates the target file
-                             // and use dpcpp as the linker tool in the
+          Tool = "$(CC) -o"; // Record the tool that generates the target file
+                             // and use 'icpx' as the linker tool in the
```
Comment (zhimingwang36) [inline clang/lib/DPCT/GenMakefile.cpp]: please test if "icpx" can link sycl object file (or, does it need icpx -fsycl).

Comment (tomflinda) [reply clang/lib/DPCT/GenMakefile.cpp]: From experiment, "icpx" cannot link sycl object file without option -fsycl

File: clang/lib/DPCT/GenMakefile.cpp (hunk 1 above)
Comment (tomflinda) [reply clang/lib/DPCT/GenMakefile.cpp]: Will update in the next updated PR.

Comment (zhimingwang36) [reply clang/lib/DPCT/GenMakefile.cpp]: Apply 'icpx -fycl' to compile the files which are originally built nvcc

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**